frontend/components/Header/Nav/Pillars/index.js

```diff
@@ -12,97 +12,98 @@ import { pillars as pillarColours } from '@guardian/pasteup/palette';
 
-const Pillars = styled('ul')({
-    clear: 'right',
-    margin: 0,
-    listStyle: 'none',
-    listStyleImage: 'none',
-    [mobileLandscape]: {
-        paddingLeft: 20,
-    },
+const pillarsStyles = css`
+    clear: right;
+    margin: 0;
+    list-style: none;
+    list-style-image: none;
+    ${mobileLandscape} {
+        padding-left: 20px;
+    }
+    li {
+        float: left;
+        ${desktop} {
+            width: 118px;
+            position: relative;
+            :after {
+                content: '';
+                display: block;
+                position: absolute;
+                right: 0;
+                top: 0;
+                bottom: 0;
+                width: 1px;
+                background-color: #abc2c9;
+            }
+        }
+        ${leftCol} {
+            width: 140px;
+        }
+    }
+`;
 
-    li: {
-        float: 'left',
-        [desktop]: {
-            width: 118,
-            position: 'relative',
-            ':after': {
-                content: '""',
-                display: 'block',
-                position: 'absolute',
-                right: 0,
-                top: 0,
-                bottom: 0,
-                width: 1,
-                backgroundColor: '#abc2c9',
-            },
-        },
-        [leftCol]: {
-            width: 140,
-        },
-    },
-});
-
-const Link = styled('a')(({ showMainMenu, pillar }) => ({
-    fontFamily: headline,
-    fontWeight: 600,
-    textDecoration: 'none',
-    color: pillarColours[pillar],
-    cursor: 'pointer',
-    display: 'block',
-    fontSize: 15.4,
-    height: 30,
-    lineHeight: 1,
-    padding: '0 4px',
-    position: 'relative',
-    overflow: 'hidden',
-    [tablet]: {
-        fontSize: 22,
-        height: 42,
-        paddingRight: 20,
-        paddingLeft: 5,
-    },
-    [desktop]: {
-        height: 48,
-    },
-    ':hover': {
-        textDecoration: showMainMenu ? 'underline' : 'none',
-    },
-    ':after': {
-        content: '""',
-        display: 'block',
-        left: 0,
-        position: 'absolute',
-        borderBottom: '4px solid currentColor',
-        right: 0,
-        bottom: -4,
-        transition: 'transform 150ms ease-out',
-    },
-    ':focus:after': {
-        transform: 'translateY(-4px)',
-    },
-    ':hover:after': {
-        transform: 'translateY(-4px)',
-    },
-}));
+const link = ({ showMainMenu, pillar }) => css`
+    font-family: ${headline};
+    font-weight: 600;
+    text-decoration: none;
+    color: ${pillarColours[pillar]};
+    cursor: pointer;
+    display: block;
+    font-size: 15.4px;
+    height: 30px;
+    line-height: 1;
+    padding: 0 4px;
+    position: relative;
+    overflow: hidden;
+    ${tablet} {
+        font-size: 22px;
+        height: 42px;
+        padding-right: 20px;
+        padding-left: 5px;
+    }
+    ${desktop} {
+        height: 48px;
+    }
+    :hover {
+        text-decoration: ${showMainMenu ? 'underline' : 'none'};
```

Comment:
We discussed whether to pull out the 'dynamic' parts of this.

Reply:
Yeah, I think this is a perfect example of a use case for extracting the dynamic styles into separate classes and using `cx` to conditionally apply them.

Reply:
Yeah, I pulled static and programmatic styles a bit further apart again. I think this is probably the right way to go here.